## Changes the GoogleGrey colors to a neutral grey

--- a/ui/gfx/color_palette.h
+++ b/ui/gfx/color_palette.h
@@ -69,16 +69,16 @@ constexpr SkColor kGoogleYellow700 = SkC
 constexpr SkColor kGoogleYellow800 = SkColorSetRGB(0xEA, 0x86, 0x00);
 constexpr SkColor kGoogleYellow900 = SkColorSetRGB(0xE3, 0x74, 0x00);
 
-constexpr SkColor kGoogleGrey050 = SkColorSetRGB(0xF8, 0xF9, 0xFA);
-constexpr SkColor kGoogleGrey100 = SkColorSetRGB(0xF1, 0xF3, 0xF4);
-constexpr SkColor kGoogleGrey200 = SkColorSetRGB(0xE8, 0xEA, 0xED);
-constexpr SkColor kGoogleGrey300 = SkColorSetRGB(0xDA, 0xDC, 0xE0);
-constexpr SkColor kGoogleGrey400 = SkColorSetRGB(0xBD, 0xC1, 0xC6);
-constexpr SkColor kGoogleGrey500 = SkColorSetRGB(0x9A, 0xA0, 0xA6);
-constexpr SkColor kGoogleGrey600 = SkColorSetRGB(0x80, 0x86, 0x8B);
-constexpr SkColor kGoogleGrey700 = SkColorSetRGB(0x5F, 0x63, 0x68);
-constexpr SkColor kGoogleGrey800 = SkColorSetRGB(0x3C, 0x40, 0x43);
-constexpr SkColor kGoogleGrey900 = SkColorSetRGB(0x20, 0x21, 0x24);
+constexpr SkColor kGoogleGrey050 = SkColorSetRGB(0xF9, 0xF9, 0xF9);
+constexpr SkColor kGoogleGrey100 = SkColorSetRGB(0xF3, 0xF3, 0xF3);
+constexpr SkColor kGoogleGrey200 = SkColorSetRGB(0xEA, 0xEA, 0xEA);
+constexpr SkColor kGoogleGrey300 = SkColorSetRGB(0xDC, 0xDC, 0xDC);
+constexpr SkColor kGoogleGrey400 = SkColorSetRGB(0xC0, 0xC0, 0xC0);
+constexpr SkColor kGoogleGrey500 = SkColorSetRGB(0xAF, 0xAF, 0xAF);
+constexpr SkColor kGoogleGrey600 = SkColorSetRGB(0x85, 0x85, 0x85);
+constexpr SkColor kGoogleGrey700 = SkColorSetRGB(0x98, 0x98, 0x98);
+constexpr SkColor kGoogleGrey800 = SkColorSetRGB(0x3F, 0x3F, 0x3F);
+constexpr SkColor kGoogleGrey900 = SkColorSetRGB(0x21, 0x21, 0x21);
 
 constexpr SkColor kGoogleOrange050 = SkColorSetRGB(0xFE, 0xEF, 0xE3);
 constexpr SkColor kGoogleOrange100 = SkColorSetRGB(0xFE, 0xDF, 0xC8);
